rcl/src/rcl/node.c

```diff
@@ -284,6 +283,20 @@ rcl_node_init(
     // error message already set
     goto fail;
   }
+  // Initialize the node type cache hash map
+  ret = rcl_node_type_cache_init(node);
```

Comment:
I think we may have lost a bug fix here, because this seems to be an issue again.  For instance, if `rcl_logging_rosout_init_publisher_for_node` fails below, then we'll leak any memory associated with the node type cache.